public/app/features/provisioning/hooks/useRepositoryAllJobs.ts

```diff
@@ -0,0 +1,73 @@
+import { HistoricJob, Job, useListHistoricJobQuery, useListJobQuery } from 'app/api/clients/provisioning';
+
+interface RepositoryHistoricalJobsArgs {
+  /** Limits the returned jobs to those which have this job name (max 1 active, unlimited historic). */
+  jobName?: string;
+  /** Limits the returned jobs to those which apply to this repository. */
+  repositoryName?: string;
+  /** Whether to continue receiving more updates of the current jobs. */
+  watch?: boolean;
+  /**
+   * How to sort the resulting jobs.
+   *
+   * - `created-first`: All jobs are treated equally. The newest jobs are shown first.
+   * - `active-first`: Active jobs are shown first, then historic jobs. Within each group, the newest jobs are shown first.
+   */
+  sort?: 'created-first' | 'active-first';
+}
+
+function labelSelectorActive(repositoryName?: string): string | undefined {
+  return repositoryName ? `repository=${repositoryName}` : undefined;
+}
+
+function labelSelectorHistoric(repositoryName?: string, jobName?: string): string | undefined {
+  const repoName = repositoryName ? `provisioning.grafana.app/repository=${repositoryName}` : '';
+  const name = jobName ? `provisioning.grafana.app/original-name=${jobName}` : '';
+
+  const selector = [repoName, name].filter(Boolean).join(', ');
+  return !!selector ? selector : undefined;
+}
+
+function fieldSelectorActive(jobName?: string): string | undefined {
+  return jobName ? `metadata.name=${jobName}` : undefined;
+}
+
+export function useRepositoryAllJobs({
```

Comment:
this is the type of code I prefer to avoid in the frontend by not having the HistoricalJob.